src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -104,7 +104,9 @@ private Assembly TryResolveAssembly(AssemblyLoadContext context, AssemblyName as
                 {
                     foreach (var extension in _extensions)
                     {
-                        var candidatePath = Path.Combine(dependencyPath, $"{assemblyName.Name}.{extension}");
+                        var candidatePath = Path.Combine(dependencyPath,
+                            assemblyName.CultureName ?? string.Empty,
```

Comment:
I think the bullet proof approach would be to add the main dependency path at the end of the current chain. It would avoid perf issues, and still hit for main assemblies with a culture on them.